From 4745ddc2b3641f408fa7f326bd72dd1113fc4eed Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 20 Sep 2019 18:22:52 +0300
Subject: [PATCH 212/245] drivers: net: phy: aquantia: fix system side protocol
 misconfiguration

commit 62c05b4e8bb967e91f51bbc184613cc64fcf423f from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

Do not set up protocols for speeds that are not supported by FW.  Enabling
these protocols leads to link issues on system side.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/phy/aquantia.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/phy/aquantia.c b/drivers/net/phy/aquantia.c
index 4a7cbe2a9ce5..5a9a292d07c9 100644
--- a/drivers/net/phy/aquantia.c
+++ b/drivers/net/phy/aquantia.c
@@ -256,10 +256,16 @@ static int aquantia_config_aneg_set_prot(struct phy_device *phydev)
         phy_write_mmd(phydev, MDIO_MMD_VEND1, AQUANTIA_VND1_GSTART_RATE,
                       aquantia_syscfg[if_type].start_rate);
 
-        for (i = 0; i <= aquantia_syscfg[if_type].cnt; i++)
+        for (i = 0; i <= aquantia_syscfg[if_type].cnt; i++) {
+		u16 reg = phy_read_mmd(phydev, MDIO_MMD_VEND1,
+                                       AQUANTIA_VND1_GSYSCFG_BASE + i);
+		if (!reg)
+			continue;
+
                 phy_write_mmd(phydev, MDIO_MMD_VEND1,
                               AQUANTIA_VND1_GSYSCFG_BASE + i,
                               aquantia_syscfg[if_type].syscfg);
+	}
 
         /* wake PHY back up */
         phy_write_mmd(phydev, MDIO_MMD_VEND1, AQUANTIA_VND1_GLOBAL_SC, 0);
-- 
2.17.1

